### DISCUSSION TITLE
custom module: how to extend existing entities

### DISCUSSION THREAD

**Discussion by swurzinger** [Blair2004/NexoPOS#1344] [Q&A]
I'd like to extend existing entities (like a user) with additional data. Therefore I create a new table with a 1:1 relationship to the existing users table and the according model. However, I'm not sure how to integrate it into the CRUD/forms system. I want to have an additional field appear e.g. on the "Edit user" page.
Can anyone help on how the hooks are supposed to be used to accomplish that?

**Comment by Blair2004** [comment]
Hi,
You can define a relation using the $relations property. See here how it's implemented

https://github.com/Blair2004/NexoPOS/blob/master/app/Crud/CustomerCrud.php#L57

**Reply by swurzinger** [reply]
Sorry, I don't understand how that answers my question. I added the relation using the `App\Crud\UserCrud@getRelations` hook, but didn't note any changes. So how does that help to display a text input field in the Edit user page?

**Reply by swurzinger** [reply]
Does anyone have any suggestion for this? Or any example code / modules I might look into? I looked into the CustomerCrud (and other CurdService) implementations, but didn't really gain valuable insights.
Unfortunately there doesn't seem to be a single module for NexoPos available on GitHub which could give some hints.
